rclcpp/include/rclcpp/subscription_base.hpp

```diff
@@ -234,6 +234,13 @@ class SubscriptionBase : public std::enable_shared_from_this<SubscriptionBase>
   void
   handle_loaned_message(void * loaned_message, const rclcpp::MessageInfo & message_info) = 0;
 
+  RCLCPP_PUBLIC
+  virtual
+  bool
+  take_loaned_message(
+    void * loaned_message,
+    const rclcpp::MessageInfo & message_info) = 0;
```

Comment:
```suggestion
    rclcpp::MessageInfo & message_info) = 0;
```

and need doc which can refer to `SubscriptionBase::take_type_erased`

Reply:
Can we implement it in SubscriptionBase without a virtual method?

Reply:
When the method is extended into subscription_base.cpp and subscription_base.hpp we won't need to make it virtual unless we extend it into the subscription.hpp and cpp files, respectively, correct? Additionally I know that we use doxygen for the cpp documentation, is there an exact place for it be rewritten for these methods?